Chroot Listener Docs (#22503)
hashicorp/vault: website/content/docs/configuration/listener/tcp.mdx

  [go-sockaddr template](https://pkg.go.dev/github.com/hashicorp/go-sockaddr/template)
  that is resolved at runtime.

> - `chroot_namespace` `(string: "admin/")` – Specifies the namespace that will be used

Reviewer: ```suggestion
- `chroot_namespace` `(string: "admin")` – Specifies the namespace that will be used
```
We don't require the trailing slash, and it's probably more natural to the user to omit it.

> as the apparent root directory by this listener. When not specified, this will default to

Reviewer: ```suggestion
as the apparent root namespace by this listener. When not specified, this will default to
```

I get the analogy with chroot(2), but I think this may confuse people.

Reviewer: Also, although I think it's implicit, it would be good to spell out what happens if the user provides a ns (i.e. we prepend the chroot ns).

> the usual "root/" namespace.

Reviewer: ```suggestion
the usual root namespace.
```

I'd rather not expose more people to this alias than we have to.  I might leave off the "when not specified" since I don't think it's necessary.

  [go-sockaddr template](https://pkg.go.dev/github.com/hashicorp/go-sockaddr/template)
  that is resolved at runtime.

> - `chroot_namespace` `(string: "admin")` – Specifies the namespace that will be used

Reviewer: ```suggestion
- `chroot_namespace` `(string: "")` – Specifies the namespace that will be used
```
I think the parenthetical value is mean to be the default, not an example.

> - `chroot_namespace` `(string: "")` – Specifies the namespace that will be used
> as the apparent root namespace by this listener. When not specified, this will default to
> the usual root namespace. If the specified namespace does not exist, requests to this
> listener will not succeed and a 400 error will be returned. If any namespace is simultaneously
> provided through either the CLI flag or the `X-Vault-Namespace` header, it will be appended
> to the `chroot_namespace` value specified here to determine the absolute namespace path.

Reviewer 2: ```suggestion
- `chroot_namespace` `(string: "")` – Specifies an alternate top-level namespace
  for the listener. Vault appends namespaces provided in CLI commands or
  `X-Vault-Namespace` headers to the top-level namespace to determine the full
  namespace path for the request. For example, if `chroot_namespace` is
  `tcp-root` and a CLI call provides the `admin` namespace, the full namespace
  path is `tcp-root/admin`. Calls to the listener will fail with a 4XX error if
  the top-level namespace provided for `chroot_namespace` does not exist.
```

I think we can simplify the description by omitting the "when unspecified" part. I think it's understood that not setting an optional configuration setting means that the product will use the defaults, and a configuration field probably isn't the right place to get into the nuances of the default namespace tree.

Author: Incorporated the changes here - changed a bit of the wording for the namespace header+cli command portion. 
Also changed the example namespaces as to not get confused.

  [go-sockaddr template](https://pkg.go.dev/github.com/hashicorp/go-sockaddr/template)
  that is resolved at runtime.

> - `chroot_namespace` `(string: "")` – Specifies an alternate top-level namespace
>   for the listener. Vault appends the namespace provided in CLI the field or
>   `X-Vault-Namespace` header to the top-level namespace to determine the full
>   namespace path for the request. For example, if `chroot_namespace` is set to
>   `admin` and the `X-Vault-Namespace` header has the value `ns1` namespace, the full
>   namespace path is `admin/ns1`. Calls to the listener will fail with a 4XX error if
>   the top-level namespace provided for `chroot_namespace` does not exist.
> 

Reviewer 2: ```suggestion
- `chroot_namespace` `(string: "")` – Specifies an alternate top-level namespace
  for the listener. Vault appends namespaces provided in the `X-Vault-Namespace`
  header or the `-namespace` field in a CLI command to the top-level namespace
  to determine the full namespace path for the request. For example, if
  `chroot_namespace` is set to `admin` and the `X-Vault-Namespace` header is
  `ns1`, the full namespace path is `admin/ns1`. Calls to the listener will fail
   with a 4XX error if the top-level namespace provided for `chroot_namespace`
   does not exist.
```

- `x_forwarded_for_authorized_addrs` `(string: <required-to-enable>)` –
  Specifies the list of source IP CIDRs for which an X-Forwarded-For header
  will be trusted. Comma-separated list or JSON array. This turns on

>   X-Forwarded-For support.  If for example Vault receives connections from the

Reviewer 2: I'm not going to add a suggestion here because it looks like you didn't actually change anything and git is just confused.
